sdk/provisioning/Azure.Provisioning/src/Infrastructure.cs

```diff
@@ -2,39 +2,142 @@
 // Licensed under the MIT License.
 
 using System;
+using System.Collections.Generic;
+using Azure.Provisioning.Expressions;
+using Azure.Provisioning.Primitives;
 
-namespace Azure.Provisioning
+namespace Azure.Provisioning;
+
+/// <summary>
+/// Collect resources and other constructs like parameters together.
+/// </summary>
+/// <param name="name"></param>
+public class Infrastructure(string name) : Provisionable
```

Comment:
`Infrastructure` is used to collect resources and other constructs.  It derives all the way from `Provisionable` at the root.  Today `Infrastructure` is kind of like a naive bicep module, but I want to blur that in the future and let people think about them as "logic groups of fun" that get written to disk according to the rules of bicep that you don't have to worry about if you don't want to _(you'll see tons of TODOs about auto-module splitting as that's not complete yet)._